src/cls/async-hooks.ts

```diff
@@ -77,15 +78,29 @@ export class AsyncHooksCLS<Context extends {}> implements CLS<Context> {
           // selection.
           this.contexts[id] = this.contexts[triggerId];
         }
-        // Note that this function always assigns values in this.contexts to
-        // values under other keys, which may or may not be undefined. Consumers
-        // of the CLS API will get the sentinel (default) value if they query
-        // the current context when it is stored as undefined.
+        // Note that this function only assigns values in this.contexts to
+        // values under other keys; it never generates new context values, and
+        // doesn't add an additional entry to this.contexts if there is no
+        // current, explicitly set context.
+        // Consumers of the CLS API will get the sentinel (default) value if
+        // they query the current context when it is not stored in
+        // this.contexts.
       },
       destroy: (id: number) => {
         // destroy is called when the AsyncResource is no longer used, so also
         // delete its entry in the map.
         delete this.contexts[id];
+      },
+      promiseResolve: (id: number) => {
+        // Some Promise async resources do not get destroyed with a destroy
```

Comment:
Is this accurate? The destroy hook will be called *eventually*, but that may be too far in the future.

The only reason why it may not get called (ever) is if the application is leaking promises.

Reply:
As far as I can tell, the common case (esp. in newer versions of Node) is that the destroy hook is not called for Promises. I'm not sure exactly which cases a destroy hook will be called, but in my own tests it seemed to get seldom, if ever, called.

Reply:
@bmeurer would this match your intuition? My understanding is that the Promise destroy hook should eventually get called unless the promise itself is leaking.

Reply:
Hm, it should always be called. It might just take a long time for the major GC to kick in.

Reply:
@kjin I would suggest experimenting with `--expose-gc` and manually calling `gloabl.gc()` (2-3 times) in strategic places to see if that changes things.

I don't think the comment is accurate as written. I still agree with the meat of this change though.

Reply:
`global.gc()` definitely changes things. I agree that the comment is inaccurate now, as there must be some condition where the destroy hook will be called. That being said it seems like as far as I've observed, it still doesn't really ever happen without explicit invocation of the gc. But that is something that should change with new versions of Node.